NeuroID/src/release/java/com/neuroid/tracker/storage/NIDDataStoreManager.kt

```diff
@@ -22,197 +21,111 @@ import org.json.JSONObject
-                val lastEvents = getStringSet(NID_STRING_EVENTS)
-                val newEvents = LinkedHashSet<String>()
-                newEvents.addAll(lastEvents)
-                newEvents.add(strEvent)
-                putStringSet(NID_STRING_EVENTS, newEvents)
-
-                var contextString: String? = ""
-                when (event.type) {
-                    SET_USER_ID -> contextString = "uid=${event.uid}"
-                    CREATE_SESSION -> contextString =
-                        "cid=${event.cid}, sh=${event.sh}, sw=${event.sw}"
-                    APPLICATION_SUBMIT -> contextString = ""
-                    TEXT_CHANGE -> contextString = "v=${event.v}, tg=${event.tg}"
-                    "SET_CHECKPOINT" -> contextString = ""
-                    "STATE_CHANGE" -> contextString = event.url ?: ""
-                    KEY_UP -> contextString = "tg=${event.tg}"
-                    KEY_DOWN -> contextString = "tg=${event.tg}"
-                    INPUT -> contextString = "v=${event.v}, tg=${event.tg}"
-                    FOCUS -> contextString = ""
-                    BLUR -> contextString = ""
-                    MOBILE_METADATA_ANDROID -> contextString = "meta=${event.metadata}"
-                    "CLICK" -> contextString = ""
-                    REGISTER_TARGET -> contextString =
-                        "et=${event.et}, rts=${event.rts}, ec=${event.ec} v=${event.v} tg=${event.tg} meta=${event.metadata}"
-                    "DEREGISTER_TARGET" -> contextString = ""
-                    TOUCH_START -> contextString = "xy=${event.touches} tg=${event.tg}"
-                    TOUCH_END -> contextString = "xy=${event.touches} tg=${event.tg}"
-                    TOUCH_MOVE -> contextString = "xy=${event.touches} tg=${event.tg}"
-                    CLOSE_SESSION -> contextString = ""
-                    "SET_VARIABLE" -> contextString = event.v ?: ""
-                    CUT -> contextString = ""
-                    COPY -> contextString = ""
-                    PASTE -> contextString = ""
-                    WINDOW_RESIZE -> contextString = "h=${event.h}, w=${event.w}"
-                    SELECT_CHANGE -> contextString = "tg=${event.tg}"
-                    WINDOW_LOAD -> contextString = "meta=${event.metadata}"
-                    WINDOW_UNLOAD -> contextString = "meta=${event.metadata}"
-                    WINDOW_BLUR -> contextString = "meta=${event.metadata}"
-                    WINDOW_FOCUS -> contextString = "meta=${event.metadata}"
-                    CONTEXT_MENU -> contextString = "meta=${event.metadata}"
-                    else -> {}
-                }
+        if (listIdsExcluded.none { it == event.tgs || it == event.tg?.get("tgs") }) {
+            val lastEventType = if (eventsList.isEmpty()) { "" } else { eventsList.last().type }
+
+            if (lastEventType == LOW_MEMORY || lastEventType == FULL_BUFFER) {
+                // no new events should be captured, drop the new event
```

Comment:
I found it useful to logEvent(eventsList.last()) here to see that the FULL_BUFFER event was found.